### PR TITLE
Changes for upgrading Babel support to 6.x.x

### PR DESCRIPTION
**Pull request by leonli**
Changes for upgrading Babel support to 6.x.x

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "url": "https://github.com/lasso-js/lasso.git"
   },
   "scripts": {
-    "test": "rm -rf .cache && node_modules/.bin/mocha --ui bdd --reporter spec ./test && node_modules/.bin/mocha --ui bdd --reporter spec ./test && node_modules/.bin/jshint lib/ taglib/"
+    "test": "rm -rf .cache && node_modules/.bin/mocha --ui bdd --reporter spec ./test && node_modules/.bin/jshint lib/ taglib/"
```
Comment (leonli) [inline package.json]: The `test` script added the mocha command twice, was it a mistake? I have no idea, so I just removed the duplicate one.

**Comment by leonli** [comment]
@patrick-steele-idem somehow, upgrading to Babel 6.x.x will cause the `out of memory` issue on running `npm test` on travis-ci, but on my Mac, it's fine. Can you please help?

**Comment by patrick-steele-idem** [comment]
Tests are passing on Travis CI now so I think we are good.

**Comment by patrick-steele-idem** [comment]
Thanks for the PR, @leonli!

**Comment by leonli** [comment]
:smile:

**Comment by laggingreflex** [comment]
+1 for babel support. But I'm still having some trouble importing `babel-polyfill`. 

I'm using babel by transpiling via gulp, and the transpiled code requires importing `babel-polyfill` at least once in your project. (earlier in babel 5 this was simpler and managed by transpiler itself I think) 

I've modified the widget-bind example a little bit: https://github.com/laggingreflex/widget-lasso-babel

I'm using an async function in `widget.es6` and importing `babel-polyfill`. Importing `babel-polyfill` is giving errors in the browser (global not defined). 

I've also tried putting `"require: babel-polyfill"` in the browser.json, which doesn't give the *same* error, but rather "regeneratorRuntime is not defined". I guess it can only be made available by directly importing babel-polyfill in the widget.es6 file itself, doing which gives the previous error. 

To compile use: `gulp build` which compiles `src/**` file to `lib/`. And to run: `node lib`

**Comment by patrick-steele-idem** [comment]
@laggingreflex Some modules are choosing to use `global` instead of `window` in the browser environment. Ideally, Lasso.js would allow `global` to work in a browser environment, but that is not there at the moment. As a simple workaround you can include some JavaScript code that does the following:

```javascript
window.global = window;
```

The babel-polyfill module should be a module that automatically runs if you were to include it in your `browser.json`:

```json
{
  "dependencies" [ "require-run: babel-polyfill" ]
}
```

Using "require-run" will just cause the module to be automatically required (and thus executed).

We are making some code improvements to Lasso.js at the moment and we should be able to fix this issues in the next version of Lasso.js.

**Pull request merged**